corehq/apps/users/views/__init__.py

```diff
@@ -710,6 +710,18 @@ def get_roles_for_display(self):
             )
         return role_view_data
 
+    def get_possible_profiles(self):
+        profile_list = []
+        definintions = CustomDataFieldsDefinition.objects.filter(domain=self.domain)
+        for definition in definintions:
+            prof_query_set = definition.get_profiles()
```

Comment:
What you're doing here works. But currently, profiles are supported only for "User fields" so I think you could also just do

```
profiles = CustomDataFieldsDefinition.get(domain, CUSTOM_USER_DATA_FIELD_TYPE).get_profiles()
```

or, this [PR](https://github.com/dimagi/commcare-hq/pull/34705/files#diff-e3f308f8032891618517b5c20eab5820702e0762f32da9be093c061f0e11bee9) adds a `get_definition_for_domain` function that you can use via `UserFieldsView.get_definition_for_domain...`

Reply:
> currently, profiles are supported only for "User fields"

I wondered about how broad this needed to be in terms of gathering profiles for the domain. `CustomDataFieldsDefinition.get_or_create` gave me pause because conceptually I don't want or need to be making changes/creating anything here, just collecting the profile options. 
I like `profiles = CustomDataFieldsDefinition.get(domain, CUSTOM_USER_DATA_FIELD_TYPE).get_profiles()`